param/parameterized.py

```diff
@@ -897,25 +898,35 @@ class ParameterMetaclass(type):
     """
     Metaclass allowing control over creation of Parameter classes.
     """
-    def __new__(mcs,classname,bases,classdict):
+    def __new__(mcs, classname, bases, classdict):
 
         # store the class's docstring in __classdoc
         if '__doc__' in classdict:
             classdict['__classdoc']=classdict['__doc__']
 
         # when asking for help on Parameter *object*, return the doc slot
-        classdict['__doc__']=property(attrgetter('doc'))
+        classdict['__doc__'] = property(attrgetter('doc'))
+
+        # Compute all slots
+        all_slots = set()
+        for base in bases:
+            for bcls in base.__mro__:
+                for slot in getattr(bcls, '__slots__', []):
```

Comment:
I think it would be faster if this `for` statement is skipped by converting directly to a set and updating `all_slots`.